### PR TITLE
feat(offcanvas): Add CSS vars for %offcanvas-css-vars + offcanvas-backdrop

### PR DESCRIPTION
**Pull request by isabellechanclou**
Add CSS vars in Offcanvas component and Docs.

To know in case:
We encounter a problem with the `toggle static offcanvas` button. A click on it didn't trigger the event and the offcanvas didn't appear. This problem appeared on macOS + Chrome/Firefox. It disappeared by itself without doing anything.

### CODE REVIEW

File: scss/_offcanvas.scss [hunk 1]
```diff
@@ -41,7 +44,7 @@
       background-clip: padding-box;
       outline: 0;
       @include box-shadow(var(--#{$prefix}offcanvas-box-shadow));
-      @include transition(transform $offcanvas-transition-duration ease-in-out);
+      @include transition(transform var(--#{$prefix}offcanvas-transition-duration) ease-in-out); // Boosted mod
```
Comment (isabellechanclou) [inline scss/_offcanvas.scss:47]: The use of css variables here shall be told to Bootstrap so that they do it on there sitde too.

Comment (louismaximepiton) [reply scss/_offcanvas.scss:47]: Yep, need to see if we update on Bs too

File: scss/_offcanvas.scss [hunk 2]
```diff
@@ -11,6 +11,9 @@
   --#{$prefix}offcanvas-border-width: #{$offcanvas-border-width};
   --#{$prefix}offcanvas-border-color: #{$offcanvas-border-color};
   --#{$prefix}offcanvas-box-shadow: #{$offcanvas-box-shadow};
+  --#{$prefix}offcanvas-zindex: #{$zindex-offcanvas}; // Boosted mod
+  --#{$prefix}offcanvas--title-line-height: #{$offcanvas-title-line-height}; // Boosted mod
```
Comment (louismaximepiton) [inline scss/_offcanvas.scss:15]: `--#{$prefix}offcanvas-title-line-height`

File: scss/_offcanvas.scss [hunk 3]
```diff
@@ -31,7 +34,7 @@
     @include media-breakpoint-down($next) {
       position: fixed;
       bottom: 0;
-      z-index: $zindex-offcanvas;
+      z-index: var(--#{$prefix}offcanvas-zindex); // Boosted mod
```
Comment (louismaximepiton) [inline scss/_offcanvas.scss:37]: Bs side too

File: scss/_offcanvas.scss [hunk 4]
```diff
@@ -133,7 +141,7 @@
 
 .offcanvas-title {
   margin-bottom: 0;
-  line-height: $offcanvas-title-line-height;
+  line-height: var(--#{$prefix}offcanvas--title-line-height); // Boosted mod
```
Comment (louismaximepiton) [inline scss/_offcanvas.scss:144]: `var(--#{$prefix}offcanvas-title-line-height);` -> Bs side too

File: scss/_offcanvas.scss [hunk 5]
```diff
@@ -114,7 +117,12 @@
 }
 
 .offcanvas-backdrop {
-  @include overlay-backdrop($zindex-offcanvas-backdrop, $offcanvas-backdrop-bg, $offcanvas-backdrop-opacity);
+  // scss-docs-start offcanvas-backdrop
```
Comment (louismaximepiton) [inline scss/_offcanvas.scss:120]: These changes may occur on Bs side too I guess

**Review by louismaximepiton — CHANGES_REQUESTED** [review]
Reading this, should maybe be on Bs side

**Review by louismaximepiton — APPROVED** [review]
Looks fine for a backport to Bootstrap

**Review by julien-deramond — CHANGES_REQUESTED** [review]

**Comment by julien-deramond** [comment]
> Looks fine for a backport to Bootstrap

Is it already proposed to Bootstrap or will you do it after the review of this PR?

**Review by julien-deramond — APPROVED** [review]
LGTM! Just changed `--#{$prefix}offcanvas-zindex-backdrop` into `--#{$prefix}offcanvas-backdrop-zindex` to have a consistent prefix.

@isabellechanclou IMO you can create a PR for Bootstrap to create the 3 CSS vars mentioned in this PR.

**Comment by isabellechanclou** [comment]
@julien-deramond : OK 👍

**Comment by isabellechanclou** [comment]
In docs/5.2/components/offcanvas/#variables, the `Added in v5.2.0` has a green (#EAF9EA) background instead of the grey (#CCCCCC) one => green color to be replaced by the grey one.

**Pull request merged**